src/network/networkd-route.c

```diff
@@ -673,10 +682,10 @@ int config_parse_gateway(const char *unit,
         if (streq(section, "Network")) {
                 /* we are not in an Route section, so treat
                  * this as the special '0' section */
-                section_line = 0;
-        }
+                r = route_new_static(network, "Network", 0, &n);
```

Comment:
When `section_line == 0`, `section` is not used at all. Please
- use NULL instead of "Network" here, we don't pass unused arguments in general
- conditionalize on `!section` in `route_new_static`, instead of `section_line == 0`.